tests/cpp/gbm/test_gbtree.cc

```diff
@@ -268,13 +273,19 @@ TEST(Dart, Prediction) {
     learner->UpdateOneIter(i, p_mat);
   }
 
+  learner->SetParam("predictor", predictor);
+
   HostDeviceVector<float> predts_training;
   learner->Predict(p_mat, false, &predts_training, 0, 0, true);
 
   HostDeviceVector<float>* inplace_predts;
-  auto adapter = std::shared_ptr<data::ArrayAdapter>(new data::ArrayAdapter{StringView{array_str}});
-  learner->InplacePredict(adapter, nullptr, PredictionType::kValue,
-                          std::numeric_limits<float>::quiet_NaN(),
+  std::shared_ptr<data::DMatrixProxy> x{new data::DMatrixProxy{}};
+  if (predictor == "gpu_predictor") {
+    x->SetData(array_str.c_str());
+  } else {
+    x->SetArrayData(array_str.c_str());
```

Comment:
It's confusing to have two functions names `SetData` and `SetArrayData`. Can we rename them to indicate that the one is for GPU and the other is for CPU?

Reply:
Renamed to SetCUDAArray.